src/main/java/johnny/Parser.java

```diff
@@ -0,0 +1,111 @@
+
+    /**
+     * Parses input String into an Arraylist
+     * of tags depending on command
+     *
+     * @return Arraylist of tags.
+     * @throws EmptyDescriptionException  If description for tasks is not mentioned.
+     * @throws NoDateException  If no/incorrectly formatted date is entered for tasks.
+     * @throws InvalidArgumentsException  If input command is invalid.
+     */
+    public ArrayList<String> parse() throws EmptyDescriptionException, NoDateException, InvalidArgumentsException {
+        String[] tags = input.split(" ", 2);
+        ArrayList<String> result = new ArrayList<>();
+
+        if(input.equals("bye")) {
+            result.add(TERMINATE);
+        } else if(input.equals("list")) {
+            result.add(PRINT_LIST);
+        } else if(input.length() >= 5 && input.substring(0, 5).equals("mark ") && isNumeric(input.substring(5))) {
+            result.add(MARK);
+            result.add(input.substring(5));
+        } else if(input.length() >= 7 && input.substring(0, 7).equals("unmark ") && isNumeric(input.substring(7))) {
+            result.add(UNMARK);
+            result.add(input.substring(7));
+        } else if(input.length() >= 7 && input.substring(0, 7).equals("delete ") && isNumeric(input.substring(7))) {
+            result.add(DELETE);
+            result.add(input.substring(7));
+        } else if(tags.length == 2 && tags[0].equals("find") && tags[1] instanceof String) {
+            result.add(FIND_EVENT);
+            result.add(tags[1]);
+        } else if(tags[0].equals("todo")) {
+            if(tags.length == 1 || tags[1].equals("")) {
+                throw new EmptyDescriptionException();
+            }
+
+            String content = tags[1];
+            if(content.equals("")) {
+                throw new EmptyDescriptionException();
+            }
+            result.add(ADD_TODO);
+            result.add(content);
+        }
+        else if(tags[0].equals("deadline")) {
+            if(tags.length == 1 || tags[1].equals("")) {
+                throw new EmptyDescriptionException();
+            }
+
+            if(!tags[1].contains("/")) {
+                throw new NoDateException();
+            }
+
+            String content = tags[1];
+            String[] details = content.split("/", 2);
+
+            result.add(ADD_DEADLINE);
+            result.add(details[0]);
+            result.add(details[1]);
+        }
+        else if(tags[0].equals("event")) {
+            if(tags.length == 1 || tags[1].equals("")) {
+                throw new EmptyDescriptionException();
+            }
+
+            if(!tags[1].contains("/")) {
+                throw new NoDateException();
+            }
+
+            String content = tags[1];
+            String[] details = content.split("/", 2);
+
+            result.add(ADD_EVENT);
+            result.add(details[0]);
+            result.add(details[1]);
+        }
+        else {
+            throw new InvalidArgumentsException(input);
+        }
```

Comment:
I like how you separate each cases clearly! However, there are missing white spaces between each "else if" and the next bracket.
In addition, some of the else/else if are not on the same line as the previous closing curly brace (such as line 89-90, 105-106).